### PR TITLE
Plugin installation (from GitHub)

### PR DESCRIPTION
**Pull request by filipgolem**
This works now:
```
# gu-hub plugins install golemfactory/golem-unlimited.
Trying to install plugins from GitHub repository: golemfactory/golem-unlimited.
Install factorization.guplug? (y/n) y
Install mining.guplug? (y/n) y
All selected plugins were installed.
```
**Warning**: GitHub REST API requires HTTPS connections. I added SSL support to Actix Web:
`actix-web = {version= "0.7", default-features=false, features = ["ssl"]}`
but it was not tested on Mac.

### CODE REVIEW

File: gu-hub/src/main.rs [hunk 1]
```diff
@@ -3,6 +3,7 @@ extern crate gu_actix;
 extern crate gu_base;
 extern crate gu_event_bus;
 extern crate gu_hardware;
+extern crate gu_hdman;
```
Comment (quilir) [inline gu-hub/src/main.rs]: I don't like import of gu_hdman in hub crate. hdman is nearly exclusively related to the provider.  I think moving out DownloadOptionsBuilder from the gu_hdman crate is the best option.

**Review by quilir — CHANGES_REQUESTED** [review]

File: gu-hub/src/main.rs (hunk 1 above)
Comment (tworec) [reply gu-hub/src/main.rs]: Any other comments?

Comment (tworec) [reply gu-hub/src/main.rs]: @filipgolem pls respond to this comment

Comment (filipgolem) [reply gu-hub/src/main.rs]: Yes, it's a good idea to move the downloader to a separate crate. This is not related to this pull request; the plugin installer uses the downloader.

**Comment by tworec** [comment]
we are postponing this PR until [actix-web v1.0](https://github.com/actix/actix-web/commits/web-v1.0.0-beta.2) will be released, because of `ssl` feature requirement

File: gu-hub/src/plugins/rest.rs [hunk 2]
```diff
@@ -53,18 +62,135 @@ pub fn install_query_inner(buf: Vec<u8>) -> impl Future<Item = (), Error = ()> {
...
+                },
+                0 => plugin_urls,
+            };
+            match plugins_to_install.len() {
+                0 => future::Either::A(future::ok(println!("No plugins were installed."))),
+                _ => {
+                    future::Either::B(
+                        ServerClient::post_json("/plug/install-github", plugins_to_install)
+                            .and_then(|r: RestResponse<InstallQueryResult>| {
+                                future::ok(debug!("{}", r.message.message()))
```
Comment (tworec) [inline gu-hub/src/plugins/rest.rs:163]:
```suggestion
                                future::ok(debug!("plugins install response: {}", r.message.message()))
```

**Review by tworec — APPROVED** [review]

**Review by prekucki — APPROVED** [review]

**Pull request merged**